planner/types.go

```diff
@@ -213,6 +216,84 @@ func (e *edges) ClearSuccessors() {
 	e.successors = e.successors[0:0]
 }
 
+func (e *edges) ClearPredecessors() {
+	e.predecessors = e.predecessors[0:0]
```

Comment:
Is there a reason not to just set this to nil?

Reply:
Well, setting it to `nil` could free the memory that backs the slice.  I think this will leave the memory around, so the slice has a length of 0 but an unchanged capacity.  So the next time a predecessor is added, no allocation will be required.  (at least I think that's how it works)